scipy/stats/_stats_py.py

```diff
@@ -3956,7 +3957,109 @@ def __init__(self, msg=None):
         self.args = (msg,)
 
 
-def pearsonr(x, y):
+def _pearsonr_fisher_ci(r, n, confidence_level, alternative):
+    """
+    Compute the confidence interval for Pearson's R.
+
+    Fisher's transformation is used to compute the confidence interval
+    (https://en.wikipedia.org/wiki/Fisher_transformation).
+    """
+    if r == 1:
+        zr = np.inf
+    elif r == -1:
+        zr = -np.inf
+    else:
+        zr = np.arctanh(r)
+    se = np.sqrt(1 / (n - 3))
+
+    if alternative == "two-sided":
+        h = special.ndtri(0.5 + confidence_level/2)
+        zlo = zr - h*se
+        zhi = zr + h*se
+        rlo = np.tanh(zlo)
+        rhi = np.tanh(zhi)
+    elif alternative == "less":
+        h = special.ndtri(confidence_level)
+        zhi = zr + h*se
+        rhi = np.tanh(zhi)
+        rlo = -1.0
+    else:
+        # alternative == "greater":
+        h = special.ndtri(confidence_level)
+        zlo = zr - h*se
+        rlo = np.tanh(zlo)
+        rhi = 1.0
+
+    return ConfidenceInterval(low=rlo, high=rhi)
+
+
+ConfidenceInterval = namedtuple('ConfidenceInterval', ['low', 'high'])
+
+PearsonRResultBase = _make_tuple_bunch('PearsonRResultBase',
+                                       ['statistic', 'pvalue'], [])
+
+
+class PearsonRResult(PearsonRResultBase):
+    """
+    Result of `scipy.stats.pearsonr`
+
+    Attributes
+    ----------
+    statistic : float
+        Pearson product-moment correlation coefficent.
+    pvalue : float
+        P-value.
+    alternative : str
+        The value of `alternative` that was passed to `pearsonr`.
+    n : int
+        The length of `x` and `y` that were given to `pearsonr`.
+
+    Methods
+    -------
+    confidence_interval(confidence_level=0.95)
+        Computes the confidence interval of the correlation
+        coefficient `statistic` for the given confidence level.
```

Comment:
How do we get this to link to the method documentation? I know that it works if I don't give a Methods section, but then it shows a bunch of other methods of `namedtuple` (e.g. `__mul__`) that aren't important here.

Reply:
Much better!

![image](https://user-images.githubusercontent.com/6570539/154796521-f016c23e-b93d-423f-9c2f-9f379addd02f.png)